modules/layers/src/solid-polygon-layer/polygon.js

```diff
@@ -287,12 +291,28 @@ export function normalize(polygon, positionSize, vertexCount) {
  * @param {Number} positionSize - size of a position, 2 (xy) or 3 (xyz)
  * @returns {Array} array of indices
  */
-export function getSurfaceIndices(normalizedPolygon, positionSize) {
+export function getSurfaceIndices(normalizedPolygon, positionSize, preproject) {
   let holeIndices = null;
 
   if (normalizedPolygon.holeIndices) {
     holeIndices = normalizedPolygon.holeIndices.map(positionIndex => positionIndex / positionSize);
   }
+  let positions = normalizedPolygon.positions || normalizedPolygon;
+
+  // TODO - handle other coordinate systems and projection modes
+  if (preproject) {
+    // When tesselating lnglat coordinates, project them to the Web Mercator plane for accuracy
+    const n = positions.length;
+    // Clone the array
+    positions = positions.slice();
```

Comment:
Could prevent some memory churn by pre-allocating this array and just setting the length here. (Also possible for holeIndices).

Reply:
`positions` is a typed array. Yes I can pre-allocate the holeIndices.